userspace/libsinsp/sinsp_filtercheck_k8s.cpp

```diff
@@ -259,69 +263,47 @@ uint8_t* sinsp_filter_check_k8s::extract(sinsp_evt *evt, OUT uint32_t* len, bool
 		break;
 	case TYPE_K8S_POD_SANDBOX_ID:
 	case TYPE_K8S_POD_FULL_SANDBOX_ID:
-		// presence of io.kubernetes.sandbox.id is enforced based on the info.sandboxID in the container status response
-		if(container_info->m_labels.count("io.kubernetes.sandbox.id") > 0)
+		// Retrieved from the ContainerStatusResponse CRI API call.
+		m_tstr = container_info->m_pod_sandbox_id;
+		if(m_field_id == TYPE_K8S_POD_SANDBOX_ID)
 		{
-			m_tstr = container_info->m_labels.at("io.kubernetes.sandbox.id");
-			if(m_field_id == TYPE_K8S_POD_SANDBOX_ID)
+			if(m_tstr.size() > 12)
 			{
-				if(m_tstr.size() > 12)
-				{
-					m_tstr.resize(12);
-				}
+				m_tstr.resize(12);
 			}
-			RETURN_EXTRACT_STRING(m_tstr);
 		}
+		RETURN_EXTRACT_STRING(m_tstr);
 		break;
 	case TYPE_K8S_POD_LABEL:
 	case TYPE_K8S_POD_LABELS:
-		if(container_info->m_labels.count("io.kubernetes.sandbox.id") > 0)
+		// Requires s_cri_extra_queries enabled, which is the default for Falco.
+		// Note that m_pod_sandbox_labels, while part of the container, is retrieved from an extra PodSandboxStatusResponse call, not the ContainerStatusResponse CRI API call.
 		{
-			std::string sandbox_container_id;
-			sandbox_container_id = container_info->m_labels.at("io.kubernetes.sandbox.id");
-			if(sandbox_container_id.size() > 12)
+			if (m_field_id == TYPE_K8S_POD_LABEL && container_info->m_pod_sandbox_labels.count(m_argname) > 0)
 			{
-				sandbox_container_id.resize(12);
+				m_tstr = container_info->m_pod_sandbox_labels.at(m_argname);
+				RETURN_EXTRACT_STRING(m_tstr);
 			}
-			const sinsp_container_info::ptr_t sandbox_container_info =
```

Comment:
No more extra pod sandbox container lookups from the container cache (now more efficient). Plus I suspect this may have caused inconsistencies in corner cases as those containers were processed at different times in the async event queue.